tests/test_codex_feed_hooks.py

```
@@ -579,6 +579,949 @@ def test_codex_monitor_survives_transient_owner_rpc_timeout(cli_path: str, root:
+        deadline = time.monotonic() + 5
+        while time.monotonic() < deadline:
+            if any(
+                frame.get("raw", "").startswith(prefix)
+                for frame in fake.frames[index:]
+            ):
+                return
+            time.sleep(0.05)
+        raise AssertionError(
+            f"bounded structured-work recovery never emitted {prefix!r}: "
+            f"{fake.frames[index:]!r}"
+        )
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)